packages/sanity/src/core/form/inputs/PortableText/hooks/useHotKeys.tsx

```diff
@@ -7,7 +7,7 @@ export function useHotkeys(hotkeys: HotkeyOptions): HotkeyOptions {
 
   // Guard that hotkeys from props will be a stable object.
   // If this props is defined inline and is always a new object, there will be issues with key handling and cursor!
-  const initialHotkeys = useMemo(() => hotkeys, [])
+  const [initialHotkeys] = useState(() => hotkeys)
```

Comment:
To save an initial snapshot of a prop it's best to use `useState`, as it's guaranteed to persist it for the lifecycle of the component.
While `useMemo` isn't guaranteed to behave this way, even though it's given an empty deps array. React reserves the right to release it to free up memory at any point in time.

Reply:
Yeah, it's important that this is only the initial prop or the warning will not work.